File: deamhome/src/main/java/sosteam/deamhome/domain/review/entity/Review.java
package sosteam.deamhome.domain.review.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.account.entity.Account;
import sosteam.deamhome.domain.item.entity.Item;
import sosteam.deamhome.global.entity.Image;
import sosteam.deamhome.global.entity.LogEntity;

import java.util.List;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review extends LogEntity {

    private String title;
    private String content;
    private int like;
    private double score;
    private boolean status;

    @DBRef(lazy = true)
    private List<Image> images;

    @DBRef(lazy = true)
    @Setter
    private Item item;

    @DBRef(lazy = true)
    @Setter
    private Account account;

    public List<Image> addImage(Image image) {
        images.add(image);

        return images;
    }
}

File: src/main/kotlin/sosteam/deamhome/domain/review/entity/Review.kt
package sosteam.deamhome.domain.review.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import sosteam.deamhome.global.entity.LogEntity

@Table("review")
class Review(
	@Id
	var id: Long?,
	var title: String,
	@Column("public_id")
	var publicId: String,
	@Column("parent_public_id")
	var parentPublicId: String,
	var content: String,
	var score: Int = 0,
	var status: Boolean = false,
	@Column("user_id")
	val userId: String,
	@Column("item_id")
	val itemId: String,
	@Column("image_urls")
	var imageUrls: MutableList<String>,
	@Column("like_users")
	var likeUsers: MutableList<String>,
	@Column("purchase_options")
	var purchaseOptions: MutableList<String>,
	@Column("report_users")
	var reportUsers: MutableList<String>,
	@Column("report_contents")
	var reportContents: MutableList<String>
) : LogEntity()
